OWML.ModHelper.Menus/ModLayoutButton.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Reflection;
+using OWML.Common.Menus;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.EventSystems;
+
+namespace OWML.ModHelper.Menus
+{
+    public class ModLayoutButton : IModLayoutButton
+    {
+        public event Action OnClick;
+
+        public Button Button { get; }
+        public IModMenu Menu { get; private set; }
+        public HorizontalLayoutGroup LayoutGroup { get; private set; }
+        private UIStyleApplier _buttonStyleApplier;
+        FieldInfo texts, foregrounds;
```

Comment:
private
Also, I've never grouped multiple variables in a single line like this anywhere else in the code base. Haven't made up my mind about it yet.

Reply:
done